src/store/multi.rs

```diff
@@ -0,0 +1,161 @@
+
+    /// Provides a cursor to all of the values for the duplicate entries that match this key
+    pub fn get<'env, T: Transaction, K: AsRef<[u8]>>(&self, txn: &'env T, k: K) -> Result<Iter<'env>, StoreError> {
+        let mut cursor = txn.open_ro_cursor(self.db).map_err(StoreError::LmdbError)?;
+        let iter = cursor.iter_dup_of(k);
+        Ok(Iter {
+            iter,
+            cursor,
+        })
+    }
+
+    /// Provides a cursor to all of the values for the duplicate entries that match this key
+    pub fn get_first<'env, T: Transaction, K: AsRef<[u8]>>(
+        &self,
+        txn: &'env T,
+        k: K,
+    ) -> Result<Option<Value<'env>>, StoreError> {
+        let result = txn.get(self.db, &k);
+        read_transform(result)
+    }
+
+    /// Insert a value at the specified key.
+    /// This put will allow duplicate entries.  If you wish to have duplicate entries
+    /// rejected, use the `put_with_flags` function and specify NO_DUP_DATA
+    pub fn put<K: AsRef<[u8]>>(&mut self, txn: &mut RwTransaction, k: K, v: &Value) -> Result<(), StoreError> {
+        let bytes = v.to_bytes()?;
+        txn.put(self.db, &k, &bytes, WriteFlags::empty()).map_err(StoreError::LmdbError)
+    }
+
+    pub fn put_with_flags<K: AsRef<[u8]>>(
+        &mut self,
+        txn: &mut RwTransaction,
+        k: K,
+        v: &Value,
+        flags: WriteFlags,
+    ) -> Result<(), StoreError> {
+        let bytes = v.to_bytes()?;
+        txn.put(self.db, &k, &bytes, flags).map_err(StoreError::LmdbError)
+    }
+
+    pub fn delete_all<K: AsRef<[u8]>>(&mut self, txn: &mut RwTransaction, k: K) -> Result<(), StoreError> {
+        txn.del(self.db, &k, None).map_err(StoreError::LmdbError)
+    }
+
+    pub fn delete<K: AsRef<[u8]>>(&mut self, txn: &mut RwTransaction, k: K, v: &Value) -> Result<(), StoreError> {
+        txn.del(self.db, &k, Some(&v.to_bytes()?)).map_err(StoreError::LmdbError)
+    }
+
+    /* TODO - Figure out how to solve the need to have the cursor stick around when
+     *        we are producing iterators from MultiIter
+    /// Provides an iterator starting at the lexographically smallest value in the store
+    pub fn iter_start(&self, store: MultiStore) -> Result<MultiIter, StoreError> {
+        let mut cursor = self.tx.open_ro_cursor(store.0).map_err(StoreError::LmdbError)?;
+
+        // We call Cursor.iter() instead of Cursor.iter_start() because
+        // the latter panics at "called `Result::unwrap()` on an `Err` value:
+        // NotFound" when there are no items in the store, whereas the former
+        // returns an iterator that yields no items.
+        //
+        // And since we create the Cursor and don't change its position, we can
+        // be sure that a call to Cursor.iter() will start at the beginning.
+        //
+        let iter = cursor.iter_dup();
+
+        Ok(MultiIter {
+            iter,
+            cursor,
+        })
+    }
+    */
+}
+
+/*
+impl<'env> Iterator for MultiIter<'env> {
```

Comment:
Nit: this comment block (and the next one) references a _MultiIter_ type, but that type isn't defined anywhere (not even in another comment block). It'd be helpful to include a declaration of that type along with this partial implementation, even if it's just a skeletal one. Or, perhaps better, move this code into an issue that tracks the problem and its resolution.